charts/nfs-subdir-external-provisioner/templates/_helpers.tpl

```diff
@@ -59,4 +59,23 @@ Return the appropriate apiVersion for podSecurityPolicy.
 {{- else -}}
 {{- print "extensions/v1beta1" -}}
 {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "nfs-subdir-external-provisioner.labels" -}}
+chart: {{ template "nfs-subdir-external-provisioner.chart" . }}
+heritage: {{ .Release.Service }}
+{{- with .Values.labels }}
+{{- toYaml . | nindent 0 }}
+{{- end }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "nfs-subdir-external-provisioner.selectorLabels" -}}
+app: {{ template "nfs-subdir-external-provisioner.name" . }}
+release: {{ .Release.Name }}
+{{- end }}
```

Comment:
Why you do not follow the official Kubernetes [label recommendations](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/)?

Also here we should take the labels which are available when creating a new chart skeleton with `helm create foo`:
```yaml
{{/*
Common labels
*/}}
{{- define "foo-chart.labels" -}}
helm.sh/chart: {{ include "foo-chart.chart" . }}
{{ include "foo-chart.selectorLabels" . }}
{{- if .Chart.AppVersion }}
app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
{{- end }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
{{- end }}

{{/*
Selector labels
*/}}
{{- define "foo-chart.selectorLabels" -}}
app.kubernetes.io/name: {{ include "foo-chart.name" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
{{- end }}
```

Reply:
i wrote in the pr description:
the default labels should have been from `app.kubernetes.io/` namespace and not just `name` or `release` by the best practices but it may breaks compatibility to someone and anyway `matchLabels` (selectorLabels) are immutable once created so i gave up on that.

Reply:
What about adding Upgrade Notes? The `--force` parameter for just this upgrade would be simple as that.

Reply:
renaming the deployment is a good solution for the immutability issue, most people not reading documentation imo. 
but what about people running kubectl in scripts with `--selector=` or anything people already did with this labels?
why to cause all that to the users when the PR intention is only to allow users add labels and annotations? i think i need a better reason why to do that especially now. best practices are the north star, not a rule...

Reply:
I thought this would not be a big issue, but for me it's okay when we leave it as-is. Maybe we integrate these changes in the next major release ;-)

When I migrated from the stable helm repository (nfs-client-provisioner) to this repo here, I also read through all values and test it on KinD or a testing cluster before I did the real migration. :-)

Reply:
that's good solution, and the default should be `legacy`. 
i am not sure that it's that important but if it makes difference to you please open separate PR if you can.